tests/validation_tests/test_failures.py

```diff
@@ -47,3 +49,24 @@ def test_serialise_user_centered_failures():
             ]
         }
     }
+
+
+def test_get_sheet_and_section():
+    failure1 = NonNullableConstraintFailure(
+        sheet="Place Details",
+        column="Question",
+    )
+    failure2 = NonNullableConstraintFailure(
+        sheet="Programme Progress",
+        column="Question",
+    )
+
+    sheet, section, column = get_sheet_section_and_column(failure1)
+    assert sheet == "Project Admin"
+    assert section == "Place Details"
+    assert column == "A1-A7"
+
+    sheet, section, column = get_sheet_section_and_column(failure2)
+    assert sheet == "Programme Progress"
+    assert section == "Programme-Wide Summary Progress"
+    assert column == "A1-A7"
```

Comment:
I think if we do go with a generic function like this then we need to test edge cases more thoroughly